packages/components/src/toolbar-group/style.scss

```diff
@@ -13,6 +15,24 @@
 	}
 
 	line-height: 0;
+
+	// Size multiple sequential buttons to be optically balanced.
+	// Icons are 36px, as set by a 24px icon and 12px padding.
+	.components-button,
+	.components-button.has-icon.has-icon {
+		min-width: $block-toolbar-height - $grid-unit-15;
+		padding-left: $grid-unit-15 * 0.5; // 6px.
```

Comment:
Would it make sense to specify this as ($button-size - $icon-size) *0.5?

Reply:
I have no idea 😬 This is based on @jasmussen's computations

Reply:
We use 6px in a bunch of places, and in most of those places we use them based on the 12px grid, which is $grid-unit-15. Hence the particular math here.

One thing we could look at is to move towards a base4 grid system instead of a base8 grid system, but this is kind of a big shift, and perhaps something to consider more for the new component system.

Reply:
Changes around here lowered the specificity of the rules for the button component, and I have a small followup to fix it in #40438.